src/libsync/syncfileitem.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) by Klaas Freitag <freitag@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "syncfileitem.h"
+#include "syncjournalfilerecord.h"
+#include "common/utility.h"
+
+#include <QLoggingCategory>
+#include "csync/vio/csync_vio_local.h"
+
+namespace OCC {
+
+Q_LOGGING_CATEGORY(lcFileItem, "sync.fileitem", QtInfoMsg)
+
+SyncJournalFileRecord SyncFileItem::toSyncJournalFileRecordWithInode(const QString &localFileName)
+{
+    SyncJournalFileRecord rec;
+    rec._path = _file;
+    rec._modtime = Utility::qDateTimeFromTime_t(_modtime);
+    rec._type = _type;
+    rec._etag = _etag;
+    rec._fileId = _fileId;
+    rec._fileSize = _size;
+    rec._remotePerm = _remotePerm;
+    rec._serverHasIgnoredFiles = _serverHasIgnoredFiles;
+    rec._checksumHeader = _checksumHeader;
+
+    // Go through csync vio just to get the inode.
+    csync_file_stat_t fs;
+    if (csync_vio_local_stat(localFileName.toUtf8().constData(), &fs) == 0) {
+        rec._inode = fs.inode;
+        qCDebug(lcFileItem) << localFileName << "Retrieved inode " << _inode << "(previous item inode: " << _inode << ")";
```

Comment:
I was about to complain that you have to destroy the `csync_file_stat_t`, but it's now value-like! +1000